crates/derive/src/stages/batch_queue.rs

```diff
@@ -243,10 +251,17 @@ where
         let origin = self.origin.ok_or(PipelineError::MissingOrigin.crit())?;
         let data = BatchWithInclusionBlock { inclusion_block: origin, batch };
         // If we drop the batch, validation logs the drop reason with WARN level.
-        if data.check_batch(&self.cfg, &self.l1_blocks, parent, &mut self.fetcher).await.is_drop() {
+        let validity =
+            data.check_batch(&self.cfg, &self.l1_blocks, parent, &mut self.fetcher).await;
+        if validity.is_drop() {
             self.prev.flush();
             return Ok(());
         }
+        // Post-Holocene, future batches are dropped due to prevent gaps.
+        if self.cfg.is_holocene_active(origin.timestamp) && validity.is_future() {
```

Comment:
same same - prob best to just combine this check i.e. `validity.is_drop() || (self.cfg.is_holocene_active(origin.timestamp) && validity.is_future())`